pennylane/transforms/tape_expand.py

```diff
@@ -159,7 +160,11 @@ def expand_fn(tape, depth=depth, **kwargs):
 
 expand_trainable_multipar = create_expand_fn(
     depth=10,
-    stop_at=not_tape | is_measurement | has_nopar | (~is_trainable) | has_gen,
+    stop_at=not_tape
+    | is_measurement
+    | has_nopar
+    | (~is_trainable)
+    | (has_gen & ~gen_is_multi_term_hamiltonian),
```

Comment:
@KetpuntoG this is also a very minor change, but I wonder if it is better to leave these two like they are (e.g., don't change `expand_multipar` and `expand_trainable_multipar`), but instead create a new one?

I'm not 100% sure. Mostly I am wondering:

- Perhaps there are other functions in PL that use these expansions
- Maybe these other functions don't mind if the generator is a Hamiltonian

@antalszava what do you think?

Reply:
That's a good point! I've checked, and it seems that only `transforms/adjoint_metric_tensor.py` uses this transform though.

> This is achieved by decomposing all trainable operations that do not have
a generator, up to maximum depth ``depth``.

Based on this part of the description, I think the current addition works: multi-term Hamiltonians don't have generators defined. So from my perspective, it should be okay.